crates/types/src/req.rs

```diff
@@ -29,11 +27,10 @@ impl SignRequest {
     pub fn signing_hash(&self) -> B256 {
         let mut hasher = Keccak256::new();
         hasher.update(DOMAIN_BINDING);
+        hasher.update(self.host_block_number.to_be_bytes::<32>());
```

Comment:
<img width="628" alt="Screenshot 2024-07-11 at 23 55 53" src="https://github.com/user-attachments/assets/dc527713-85fb-4653-a94d-49a25885479b">

should match ordering in the contract

https://github.com/init4tech/zenith/blob/main/src/Zenith.sol#L131-L142